File: ovh-monitor.js
const axios = require("axios");
const https = require("https");

// ---------------- SETTINGS ----------------
const sleepTime = 600000; // 10 minutes
const planCode = "vps-2025-model1";
const datacenterCode = "ap-southeast-sgp";

const TELEGRAM_TOKEN = "";
const CHAT_ID = ""; // your personal user ID

// ---------------- FORCE IPV4 ----------------
const agent = new https.Agent({
    family: 4
});

// ---------------- HELPERS ----------------
function sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
}

async function sendTelegram(message) {
    try {
        const url = `https://api.telegram.org/bot${TELEGRAM_TOKEN}/sendMessage`;

        const response = await axios.post(
            url,
            {
                chat_id: CHAT_ID,
                text: message
            },
            {
                httpsAgent: agent,
                timeout: 10000
            }
        );

        console.log("Telegram sent:", response.data.ok);

    } catch (err) {
        console.log("Telegram error:", err.response?.data || err.message);
    }
}

// ---------------- MAIN CHECK ----------------
async function checkVPS() {
    try {
        console.log("Checking OVH SG VPS...");
        const now = new Date().toLocaleString();
        console.log("Time:", now);
        console.log("----------------------------------");

        const url = `https://ca.api.ovh.com/v1/vps/order/rule/datacenter?ovhSubsidiary=SG&planCode=${planCode}`;

        const response = await axios.get(url, {
            timeout: 10000
        });

        const datacenters = response.data.datacenters;

        const sg = datacenters.find(
            dc => dc.code === datacenterCode || dc.datacenter === "SGP"
        );

        if (!sg) {
            await sendTelegram(`⚠️ SG datacenter not found at ${now}`);
            return;
        }

        const linuxStatus = sg.linuxStatus;
        const windowsStatus = sg.windowsStatus;

        console.log("Linux:", linuxStatus);
        console.log("Windows:", windowsStatus);

        const baseMessage =
`📡 OVH SG VPS Check
🕒 ${now}

Linux: ${linuxStatus}
Windows: ${windowsStatus}`;

        // 🚨 ALERT MODE
        if (linuxStatus === "available") {

            console.log("🚨 LINUX AVAILABLE — STARTING ALERT SPAM");

            for (let i = 1; i <= 5; i++) {

                await sendTelegram(
`🚨🚨 LINUX VPS AVAILABLE 🚨🚨

Alert ${i}/5

${baseMessage}`
                );

                if (i < 5) {
                    await sleep(30000); // 30 seconds
                }
            }

        } else {
            // Normal update
            await sendTelegram(baseMessage);
        }

    } catch (err) {
        console.log("Main error:", err.message);
    }
}

// ---------------- SAFE LOOP ----------------
async function startMonitor() {
    await checkVPS();
    setTimeout(startMonitor, sleepTime);
}

startMonitor();

File: ovh_monitor.py
import requests
import time
import os
from datetime import datetime

# ---------------- SETTINGS ----------------
SLEEP_TIME = 600  # 10 minutes in seconds
PLAN_CODE = "vps-2025-model1"
DATACENTER_CODE = "ap-southeast-sgp"

# Use environment variables for security
TELEGRAM_TOKEN = os.getenv("TELEGRAM_TOKEN", "")
CHAT_ID = os.getenv("CHAT_ID", "")

def send_telegram(message):
    if not TELEGRAM_TOKEN or not CHAT_ID:
        print("Error: Missing Telegram credentials.")
        return

    url = f"https://api.telegram.org/bot{TELEGRAM_TOKEN}/sendMessage"
    payload = {
        "chat_id": CHAT_ID,
        "text": message
    }
    
    try:
        # family=4 equivalent: most modern systems handle this, 
        # but you can use a custom adapter if you strictly need IPv4 forcing.
        response = requests.post(url, json=payload, timeout=10)
        print(f"Telegram sent: {response.json().get('ok')}")
    except Exception as e:
        print(f"Telegram error: {e}")

def check_vps():
    try:
        print("Checking OVH SG VPS...")
        now = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
        print(f"Time: {now}")
        print("-" * 34)

        url = f"https://ca.api.ovh.com/v1/vps/order/rule/datacenter?ovhSubsidiary=SG&planCode={PLAN_CODE}"
        
        response = requests.get(url, timeout=10)
        response.raise_for_status()
        
        datacenters = response.json().get("datacenters", [])
        
        # Find Singapore datacenter
        sg = next((dc for dc in datacenters if dc.get("code") == DATACENTER_CODE or dc.get("datacenter") == "SGP"), None)

        if not sg:
            send_telegram(f"⚠️ SG datacenter not found at {now}")
            return

        linux_status = sg.get("linuxStatus")
        windows_status = sg.get("windowsStatus")

        print(f"Linux: {linux_status}")
        print(f"Windows: {windows_status}")

        base_message = (
            f"📡 OVH SG VPS Check\n"
            f"🕒 {now}\n\n"
            f"Linux: {linux_status}\n"
            f"Windows: {windows_status}"
        )

        # 🚨 ALERT MODE
        if linux_status == "available":
            print("🚨 LINUX AVAILABLE — STARTING ALERT SPAM")
            for i in range(1, 6):
                alert_msg = f"🚨🚨 LINUX VPS AVAILABLE 🚨🚨\n\nAlert {i}/5\n\n{base_message}"
                send_telegram(alert_msg)
                
                if i < 5:
                    time.sleep(30)  # 30 seconds
        else:
            # Normal update
            send_telegram(base_message)

    except Exception as e:
        print(f"Main error: {e}")

def start_monitor():
    while True:
        check_vps()
        print(f"Waiting {SLEEP_TIME//60} minutes for next check...")
        time.sleep(SLEEP_TIME)

if __name__ == "__main__":
    start_monitor()
